test/test_torch.py

```diff
@@ -17272,6 +17272,60 @@ def test_reciprocal_complex_extremal(self, device, dtype):
 
         self.compare_with_numpy(torch.reciprocal, np.reciprocal, vals, device, dtype)
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.bool, *(torch.testing.get_all_int_dtypes() + torch.testing.get_all_fp_dtypes()))
```

Comment:
@dtypes supports iterables of tuples of dtypes, too, in which case it will pass the tuple of dtypes instead of the regular "dtype" argument. This pattern will avoid you have to enumerate over the dtypes again in the function below and improve test granularity.